File: src/main/java/com/dv/telegram/WikiPageData.java
package com.dv.telegram;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.apache.commons.lang3.StringUtils;

import java.util.List;

@Data
@AllArgsConstructor
public class WikiPageData {
    private String name;
    private String url;
    private String wordsString;
    private List<String> words;

    public boolean isPresentIn(String text) {
        if (StringUtils.isBlank(text)) {
            return false;
        }

        for (String word : words) {
            if (text.contains(word)) {
                return true;
            }
        }

        return false;
    }

    public String getOneLineAnswer() {
        return String.format("%s — %s", name, url);
    }

    public String getMultiLineAnswer() {
        return String.format("— %s — %s", name, url);
    }
}

File: src/main/kotlin/com/dv/telegram/data/WikiPageData.kt
package com.dv.telegram.data

data class WikiPageData(
    val name: String,
    val url: String,
    val wordsString: String,
    val words: List<String>
) : BotAnswerData {
    override fun isPresentIn(text: String) = isPresentIn(text, words)

    override fun getMatch(text: String) = getMatch(text, words)

    fun getOneLineAnswer() = "$name — $url"

    fun getMultiLineAnswer() = "▫ $name — $url" // todo: use configurable bullet
}
